### DISCUSSION TITLE
Problem with creating measures for unique values in a column

### DISCUSSION THREAD

**Discussion by SimonAnalyst** [TabularEditor/TabularEditor3#1417] [🙋♀️TE3 Q&A]
Hello,

I am trying to create measures dynamically in Tabular Editor for each unique value in the column FactTable[Prod Realization]. I want to create a measure that sums the values in FactTable[ActualsYTD], with a filter applied for each unique value in FactTable[Prod Realization].

The goal is to generate a measure for each unique value in FactTable[Prod Realization], calculating the sum of FactTable[ActualsYTD] filtered by that value.

Can anyone point out what might be going wrong or suggest a solution to ensure only the correct unique values generate measures?

Thanks in advance!

**Comment by otykier** [comment]
Am I assuming correctly that you want to create a C# script to do this? I'm curious to hear what you've tried so far.

In any case, something like the following should work:

```csharp
var dataReader = ExecuteReader("EVALUATE DISTINCT(FactTable[Prod Realization])");
while(dataReader.Read())
{
    var value = dataReader.GetString(0);
    Model.Tables["FactTable"].AddMeasure(
        // Name of measure:
        "Realization " + value,
        // DAX expression:
        "CALCULATE(SUM([FactTable[ActualsYTD]), FactTable[Prod Realization] = \"" + value + "\")"
    );
}
```

I tested the script on the AdventureWorks model, using the `Product[Color]` column. It created the following measures:

![image](https://github.com/user-attachments/assets/b6b61693-2054-4266-ae52-94b99fe9fb71)
